ignf_gpf_sdk/workflow/action/ProcessingExecutionAction.py

```diff
@@ -199,75 +200,95 @@ def find_stored_data(self, datastore: Optional[str] = None) -> Optional[StoredDa
-            # Si on est sorti c'est que c'est fini
-            ## dernier affichage
-            if callback is not None:
-                callback(self.processing_execution)
-            ## on return le status de fin
-            return str(s_status)
-        except KeyboardInterrupt as e:
-            # TODO
-            # si le traitement est déjà dans un statu fini on ne fait rien => transmission de l'interruption
-            self.processing_execution.api_update()
-            s_status = self.processing_execution.get_store_properties()["status"]
-            if s_status in [ProcessingExecution.STATUS_ABORTED, ProcessingExecution.STATUS_SUCCESS, ProcessingExecution.STATUS_FAILURE]:
-                Config().om.warning("traitement déjà fini")
-                raise
-            # arrêt du traitement
-            Config().om.warning("Ctrl+C : traitement en cours d’interruption, veuillez attendre...")
-            self.processing_execution.api_abort()
-            # attente que le traitement passe dans un statu fini
-            self.processing_execution.api_update()
-            s_status = self.processing_execution.get_store_properties()["status"]
-            while s_status not in [ProcessingExecution.STATUS_ABORTED, ProcessingExecution.STATUS_SUCCESS, ProcessingExecution.STATUS_FAILURE]:
-                # On attend 2s
-                time.sleep(2)
-                # On met à jour __processing_execution + valeur status
-                self.processing_execution.api_update()
-                s_status = self.processing_execution.get_store_properties()["status"]
-            ## dernier affichage
-            if callback is not None:
-                callback(self.processing_execution)
-            if s_status == ProcessingExecution.STATUS_ABORTED and self.output_new_entity:
-                # suppression de l'upload ou la stored data en sortie
-                if self.upload is not None:
-                    Config().om.warning("Suppression de l'upload en cours de remplissage suite à l’interruption du programme")
-                    self.upload.api_delete()
-                elif self.stored_data is not None:
-                    Config().om.warning("Suppression de la stored-data en cours de remplissage suite à l'interruption du programme")
-                    self.stored_data.api_delete()
-                # transmission de l'interruption
-            raise KeyboardInterrupt() from e
+
+            except KeyboardInterrupt as e:
+                # on appelle la callback de gestion du ctrl-C
+                if ctrl_c_action is None or ctrl_c_action():
+                    # on arrêter le traitement :
+                    # si le traitement est déjà dans un statut terminé, on ne fait rien => transmission de l'interruption
+                    self.processing_execution.api_update()
+                    s_status = self.processing_execution.get_store_properties()["status"]
```

Comment:
C'est juste étonnant d'avoir la mise à jour après le commentaire "si le traitement est déjà..."
```suggestion
                    # on arrêter le traitement : (maj + action spécifique selon le statut)

                    # mise à jour du traitement
                    self.processing_execution.api_update()

                    # si le traitement est déjà dans un statut terminé, on ne fait rien => transmission de l'interruption
                    s_status = self.processing_execution.get_store_properties()["status"]
```